### PR TITLE
docs: add a guide for local vs remote clusters

### PR DESCRIPTION
**Pull request by nicks**
Hello @jazzdan, @victorwuky,

Please review the following commits I made in branch nicks/remote:

444905d2b853729ce3c5357e9774e8ab3d4f69f3 (2020-06-11 14:14:27 -0400)
docs: add a guide for local vs remote clusters

Code review reminders, by giving a LGTM you attest that:

* Commits are adequately tested
* Code is easy to understand and conforms to style guides
* Incomplete code is marked with TODOs
* Code is suitably instrumented with logging and metrics

### CODE REVIEW

**Comment by nicks** [comment]
I feel like we've answered this question enough times (across github issues and slack) that I was able to do this by copying-and-pasting things we've written in other places and massaging it a bit

File: docs/local_vs_remote.md [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+- Communcation between the build jobs and your cluster image registry
+
+- Caching builds effectively
+
+If you're interested in setting this up, please reach out to us.
+
+[Schedule a Meeting](https://calendly.com/nick-at-tilt/remote-builds)
+
+We are considering native solutions in Tilt to auto-configure this,
+and can share how it might work with your setup.
```
Comment (jazzdan) [inline docs/local_vs_remote.md:124]: This part reads a little stilted to me, maybe:

```markdown
If you're interested in setting this up, or chatting about potential native solutions in Tilt to auto-configure this, please [reach out to us](https://calendly.com/nick-at-tilt/remote-builds).
```

File: docs/local_vs_remote.md [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+  clusters](https://istio.io/latest/docs/setup/install/multicluster/).
+
+- [Inlets](https://blog.alexellis.io/ingress-for-your-local-kubernetes-cluster/)
+  allocates a public load balancer to route traffic to your local
+  cluster.
+
+- [Telepresence](https://telepresence.io) replaces your remote service
+  with a proxy that routes traffic to your local service.
+  
+We think these projects are great and complement Tilt well! You can use them o
```
Comment (jazzdan) [inline docs/local_vs_remote.md:143]:
```suggestion
We think these projects are great and complement Tilt well! You can use them to
```

File: docs/local_vs_remote.md (hunk 1 above)
Comment (nicks) [reply docs/local_vs_remote.md]: +1, fixed

File: docs/local_vs_remote.md (hunk 2 above)
Comment (nicks) [reply docs/local_vs_remote.md]: done

**Pull request merged**